### DISCUSSION TITLE
NBMiner v42.2

### DISCUSSION THREAD

**Discussion by nebutech-admin** [NebuTech/NBMiner#896] [General]
- `feature`: `ethash` Add a new option `--lhr-mode` to select LHR unlock mode.
  - `--lhr-mode 1` is the default mode and is the same as which in `v41.5`
  - Try to use `--lhr-mode 2` if stability issue encountered in `mode 1`, for json config, use `"lhr-mode": "2"`
  - `LHR v3` GPUs can only use `mode 1`
- `feature`: `ethash` Extend LHR unlocker to work with old driver versions. Starting from `Windows v512.95` and `Linux v515.x`, LHR unlocker no longer works, make sure not to use these new driver versions.
- `optimize`: `ethash` small hashrate improvement on Nvidia GPUs.
- `feature`: Add `GDDR6X` memory temp in summary table for Linux.
- `feature`: Add `GPU RAM type` and `GPU RAM vendor` in log.
- `feature`: Add Nvidia driver versoin in summary table.
- `feature`: `ethash` Add `eth_submitHashrate` for `nicehash (EthereumStratum)` protocol.
- `feature`: If `Overclock` applies through `nbminer`, reset settings to stock when miner exit.
- `feature`: `-mt` option on windows does not require custom driver anymore (admin priviledge is required).
- **`NOTE`: NBMiner versions older than v39.5 (not included) will run into issue when ETH reaches `EPOCH 520` (Approx mid September). Please make sure to upgrade to a newer version before that if ETH is still with POW.**

<hr /><em>This discussion was created from the release <a href='https://github.com/NebuTech/NBMiner/releases/tag/v42.2'>NBMiner v42.2</a>.</em>

**Comment by droomagon** [comment]
question about the drivers version,  (Starting from Windows v512.95 and Linux v515.x)

does this release (42.2) only work with 512.95, but not 512.15 ?

**Reply by Oleg1606** [reply]
512.15 работает

**Reply by YashvirGaming** [reply]
luckily i installed 512.77 so it should work for me
otherwise i will have to re use that DDU uninstaller clean perform gpu drivers

**Comment by danieldobrea** [comment]
Hi! What is the commad to update with this version in hiveos? Thank you!
L.E. I founded:
cd /tmp && wget https://github.com/NebuTech/NBMiner/releases/download/v42.2/NBMiner_42.2_Linux.tgz && tar -xvf NBMiner_42.2_Linux.tgz && cd NBMiner_Linux && miner stop && cp nbminer /hive/miners/nbminer/42.1 && miner start

**Comment by Ivan-bykatyj** [comment]
Нб майнер последний нестабильный,срабатывают скорее всего локи, и хэш незначительно прыгает...

**Comment by YoRu-H** [comment]
Tuve un error dos veces despues de no aceptar procesos por unos 30 minutos, quedo en pausa en vez de reiniciarse como lo hacia en versiones anteriores. despues de precionar cualquier tecla para continuar el minero se cierra. como lo cambio para que reinicie solo la mineria cuando hay error?

**Comment by lsj13145222** [comment]
I use the “-cclock @1500” command,But when I quit bat,  the CClock is still fixed at 1500,MClock will recover automatically

**Reply by DAVIARY** [reply]
yea it's not working correctly

**Comment by hushpuppyminer01** [comment]
On Test  Nbminer Version 42.2 
My Card 3060 v1 
ND 512.15 
Hashrate 48.8 Max

![Screenshot (126)](https://user-images.githubusercontent.com/105936076/170823127-95c638f9-cebc-45da-8d48-2b6a328ec36e.png)

**Reply by Oleg1606** [reply]
47.5 - 50.68  3060 v2   дрова 512.15 win10
Все зависит от карты.

**Reply by droomagon** [reply]
running max at 48.8? not very efficient though..

**Reply by acidstout** [reply]
48.8 @ 120W with quite high mem-clock and variable core-clock is definitely not efficient. A high hashrate is not always achieved by overclocking your card. You have to experiment with different settings, and with regards to increasing electricity prices you should keep an eye on the overall power consumption of your rig.

These are my specs. Maybe they help you achieving more efficiency.
NBMiner 41.5 (command-line parameters are "-a ethash -log -pl 100 -fan 60")
Geforce 3060Ti v1 (non-LHR)
Drivers 512.15 with NVENC and NvFBC patches applied (shouldn't affect the hashrate)

If I set power consumption to unlimited, the hashrate increases to appr. 54 MH/s, but power consumption will increase to >120W. So, better limit the power consumption.
![image](https://user-images.githubusercontent.com/23167347/186898899-3093a879-e7fa-4ddd-9e71-952a7b0f3c62.png)

**Reply by YashvirGaming** [reply]
> 48.8 @ 120W with quite high mem-clock and variable core-clock is definitely not efficient. A high hashrate is not always achieved by overclocking your card. You have to experiment with different settings, and with regards to increasing electricity prices you should keep an eye on the overall power consumption of your rig.
> 
> These are my specs. Maybe they help you achieving more efficiency. NBMiner 41.5 (command-line parameters are "-a ethash -log -pl 100 -fan 60") Geforce 3060Ti v1 (non-LHR) Drivers 512.15 with NVENC and NvFBC patches applied (shouldn't affect the hashrate)
> 
> If I set power consumption to unlimited, the hashrate increases to appr. 54 MH/s, but power consumption will increase to >120W. So, better limit the power consumption. ![image](https://user-images.githubusercontent.com/23167347/186898899-3093a879-e7fa-4ddd-9e71-952a7b0f3c62.png)



Hey brother can you tell me if this GOOD OR BAD
Should I be getting more hashrate here or not 
Also if the GPU temps is correct or borderline (overheating) or not
![is this good or bad](https://user-images.githubusercontent.com/64865096/186915507-e545214b-ba62-481a-b6de-f06ad35be0ce.png)

**Reply by acidstout** [reply]
With a 3070 you should get a slightly higher hashrate than me. This also depends on the driver version, the version of the card (LHR or non-LHR; I don't have hashrates for that card as reference, but Google should be helpful) and on other things like apps running in the background using your GPU. For instance I use Lively (animated wallpaper), which - if enabled - reduces my hashrate to about 49MH/s. The same happens when watching a movie, playing games, ... The reason for this is, that NBMiner seems to have a lower priority than other apps which use 3D-acceleration. E.g., I can run GTA V perfectly smooth in 4K on my card while NBMiner runs, but the hashrate will drop to zero.

Try reducing the  mem-clock. 7800 is much too high. I would reduce it to 7000 or maybe 7200, but not more. Then increase the core-clock in small steps by 100MHz. Experiment with the settings in MSI Afterburner to find the sweet spot.

On my card I get better hashrates without overclocking.

**Reply by YashvirGaming** [reply]
Check it out new OC Settings 

![Untitled](https://user-images.githubusercontent.com/64865096/187048151-8d9b13a0-1b3a-4af5-9783-35e6426ddfab.png)

**Reply by acidstout** [reply]
That's some nice hashrate now. Big improvement to your previous one. What exactly did you change?

**Reply by YashvirGaming** [reply]
well after spent hours researching and youtubing around 
but thats it im stuck on 57-59 Mh/s it won't get 60-64Mh/s
How to get 64Mh/s on Ethereum mining 
I am looking for that one cannot find any settings for that or video.
I guess to reach that 64Mh you really need a founders edition GPU 
What do you think brother?

**Comment by nbminer-beta** [comment]
lhr mode 2 makes bug on 1660 super cards. (power limit bug and cclock bug )

I noticed it after 7 hours its burned my 1660s if noticed more hours later.

Normaly its running with 52C maximum standart mining with mode 2 its hit pl 100 and cclock to 1830 and it was running on 79C.

.

**Reply by Oleg1606** [reply]
Такое происходит на любом майнере и любой версии - карта переходит на стоковый разгон. Правильная настройка кривой управлением кулеров позволит избежать фатального  перегрева карты ( в большинстве случаев ).

**Comment by Marsi4eg** [comment]
What is the oldest supported driver? Had someone tried this on Win7? The last driver version for it is 473.62

On 461.92 the miner doesn't even start in ethash mode. 3060 v1. Miner versions prior to full LHR unlock work good.

**Reply by Volterier** [reply]
Win 7 + 473.04 working  wel.

**Comment by DAVIARY** [comment]
feature: If Overclock applies through nbminer, reset settings to stock when miner exit.
This is not working correctly. Doesn't reset to default

**Comment by rexanoob** [comment]
hi guys
i have 2 rtx 3080 10g rev 1.0
my hashrate after the start miner is 90mh and core is 1800 (display on afterburner or gpu-z sensor tab)
after few second , hashrate down to 61mh and core is 650 (display on afterburner or gpu-z sensor tab)
i tried 
any miner and nbminer.
how can i fix this problem?
![hs](https://user-images.githubusercontent.com/108605774/177029585-0fae5c9a-2dc7-4e97-91ff-de76c0e609ce.png)

**Reply by Oleg1606** [reply]
Для начала снизь PL до незначительного снижения хеша - лишняя мощность  не добавит выхлопа. Вариант 95% кулера не есть хорошее решение - это убой кулеров. Почитай на форумах о теме.

**Reply by Oleg1606** [reply]
Снизь разгон  МС - может быть переразгон - тогда при запуске большой хэш но постепенно снижается.

**Reply by rexanoob** [reply]
breaking news! :D i set the oc default . i see my memory temp on sensor tab(gpu-z) before starting miner is 50c .  after starting miner , memory temp up to 110c :-O . why? 

memory temp 50c = core clock 1800mhz
memory temp 108c = core clock down to 600mhz
my oc:
core clock : 0
memory clock : 0
power limit : 85
temp limit : 80

**Reply by Tsakalos** [reply]
Go to curve editor. Select a cclock near to 1300-1400MHz and press ctrl+L. Apply the changes and thank me later. Cheers.

**Reply by YashvirGaming** [reply]
> breaking news! :D i set the oc default . i see my memory temp on sensor tab(gpu-z) before starting miner is 50c . after starting miner , memory temp up to 110c :-O . why?
> 
> memory temp 50c = core clock 1800mhz memory temp 108c = core clock down to 600mhz my oc: core clock : 0 memory clock : 0 power limit : 85 temp limit : 80


This video will help you
https://youtu.be/oCyxZASezWY

**Comment by AndreJoker** [comment]
подскажите как включить отображение сложности шар в общей статистике

**Reply by YashvirGaming** [reply]
Watch video: [https://youtu.be/oCyxZASezWY](https://youtu.be/oCyxZASezWY)

**Comment by Tsakalos** [comment]
Is there a way to reduce temps in 3080lhr? I hit around 98-102C. Thanks in advance.

**Reply by YashvirGaming** [reply]
This video might help you
https://youtu.be/oCyxZASezWY

**Comment by wezunchik44** [comment]
tell me how to disable one of the video cards, I can’t enter commands, where to enter them correctly?
original comment: подскажите как отключить одну из видеокарт , ввод команд не получается у меня, куда их правильно вводить?

**Reply by ilbash** [reply]
CUDA_VISIBLE_DEVICES

**Reply by YashvirGaming** [reply]
> CUDA_VISIBLE_DEVICES

**### help here why it doesn't wanna Benchmark** 
![image](https://user-images.githubusercontent.com/64865096/186916577-3736045d-5287-420d-9820-38170a678072.png)

**Comment by s0laria** [comment]
Driver 510.85 halves the hashrate on 3060Ti, at least on my Ubuntu 18.04 miners. One has to downgrade back to 510.73 installing the driver runfile from nvidia webpage, since 510.73 is no longer available in Ubuntu repositories!
